controller.go

```diff
@@ -284,13 +284,17 @@ func syncPV(pv *PV) {
 			// NOTE: releasePV may either release the PV back into the pool or
 			// recycle it or do nothing (retain)
 
-			// HOWTO RELEASE A PV
-			pv.Status.Phase = Released
-			if err := CommitPVStatus(pv); err != nil {
-				// Status was not saved; we will fall back into the same
-				// condition in the next call to this method
-				return
+			// Make sure we don't overwrite previous failure of
+			// recycler/deleter.
+			if pv.Status.Phase != Failed {
```

Comment:
So, I've now typed 3 versions of this comment and lost them through different ways.  Let's see if I can make this one stick.

I think @swagiaal is right to ask the question he does, and I also think what this PR is doing is thematically correct.  The discomfort is that we agreed as a group to do no comparisons on status or phase.  I think that is a good thing to do, so what about this?  Annotate the volume as failed is the recycler fails -- no need to switch on status.

Thoughts?  @jsafrane @swagiaal